Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -1352,7 +1349,18 @@ CompletionHandler<void(Decoder*)> takeAsyncReplyHandler(Connection& connection,
 
 void Connection::wakeUpRunLoop()
 {
-    RunLoop::main().wakeUp();
+    if (!isValid())
+        return;
+    m_syncState->runLoop().wakeUp();
+}
+
+template<typename F>
+void Connection::dispatchClient(F&& clientRunLoopTask)
```

Comment:
It might be nice to have `dispatchClientLocked` that this calls into, and use that for the other call sites that currently call `m_syncState->runLoop().dispatch()`. Just makes it easier to see from a glance that the differing behaviour is due to whether we have the lock already.

Reply:
Good point, I'll consider it.
There are 2 differences:
 - locked or not
 - connection known to be valid or not (we have runloop or we don't know if we have client runloop or not)

So above in Connection::dispatchIncomingMessages(), if we would call dispatchClientLocked(), but that would be 1% confusing as we know that "dispatch to client locked" does "if connection is valid, send to client", and in that function we know that connection must be valid for it to execute at that point. So then in that function, we'd stop to think "I thought the connection is valid here. Am I missing something when we dispatch with a function that checks for the validity?".